name: Test consuming this action
on:
  release:
    types: [released]
  workflow_run:
    workflows: ["Release"]
    types: ["completed"]
jobs:
  test_success:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: ./

  test_error:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - id: test
        continue-on-error: true
        uses: ./
        with:
          error: "This is an error"
      - name: Verify failure
        if: steps.test.outputs.error != ''
        run: echo "Failed as expected"
      - name: Unexpected success
        if: steps.test.outputs.error == ''
        run: echo "Succeeded unexpectedly" && exit 1
